guides/nodejs.md

```diff
@@ -0,0 +1,308 @@
+# Node.js Developer's Guide
+
+When creating a Node.js function using the `func` binary, the project directory
+looks like a typical Node.js project. Both HTTP and Event functions have the same
+template structure.
```

Comment:
And I'd like CLI reference guide in here as well, it could be hard for newcomers to find it (especially if it is in another repo).

https://github.com/boson-project/func/blob/main/docs/commands.md#cli-commands